relay-server/src/actors/global_config.rs

```diff
@@ -0,0 +1,105 @@
+use std::borrow::Cow;
+use std::time::Duration;
+
+use relay_dynamic_config::GlobalConfig;
+use relay_statsd::metric;
+use relay_system::{Addr, Service};
+use reqwest::Method;
+use serde::{Deserialize, Serialize};
+
+use crate::actors::processor::EnvelopeProcessor;
+use crate::actors::upstream::{RequestPriority, SendQuery, UpstreamQuery, UpstreamRelay};
+use crate::statsd::RelayCounters;
+
+/// Service implementing the [`GlobalConfig`] interface.
+///
+/// The service is responsible for fetching the global config and
+/// forwarding it to the services that require it.
+#[derive(Debug)]
+pub struct GlobalConfigService {
+    envelope_processor: Addr<EnvelopeProcessor>,
+    upstream: Addr<UpstreamRelay>,
+}
+
+impl GlobalConfigService {
+    pub fn new(envelope_processor: Addr<EnvelopeProcessor>, upstream: Addr<UpstreamRelay>) -> Self {
+        Self {
+            envelope_processor,
+            upstream,
+        }
+    }
+
+    /// Forwards the given global config to the services that require it.
+    fn update_global_config(&mut self) {
+        let upstream_relay: Addr<UpstreamRelay> = self.upstream.clone();
+        let envelope_processor = self.envelope_processor.clone();
+        tokio::spawn(async move {
+            let query = GetGlobalConfig { global_config: () };
+
+            if let Ok(Ok(response)) = upstream_relay.send(SendQuery(query)).await {
+                metric!(counter(RelayCounters::GlobalConfigFetchSuccess) += 1);
+                envelope_processor.send::<GlobalConfig>(response.global);
+            } else {
+                metric!(counter(RelayCounters::GlobalConfigFetchFailed) += 1);
+            }
+        });
+    }
+}
+
+impl Service for GlobalConfigService {
+    type Interface = ();
+
+    fn spawn_handler(mut self, _rx: relay_system::Receiver<Self::Interface>) {
+        tokio::spawn(async move {
+            let mut ticker = tokio::time::interval(Duration::from_secs(10));
+            relay_log::info!("global config started");
+
+            loop {
+                tokio::select! {
+                    biased;
+                    _ = ticker.tick() => self.update_global_config(),
+                    else => break,
+                }
+            }
+            relay_log::info!("global config stopped");
+        });
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GetGlobalConfig {
+    // This struct is only used for getting the global configs, meaning the presence of the key
+    // is enough, and no value needs to be included.
+    global_config: (),
+}
```

Comment:
is my logic here correct? if we only send this request for global_config, then it is redundant to have something like global_config = true, iiuc sentry will receive the serialized version of this, which will probably have "global_config": null, as the request body, and we only check if the global_config key is present in that case